src/audio/module_adapter/module/generic.c

```diff
@@ -288,10 +288,14 @@ int module_process_sink_src(struct processing_module *mod,
 	assert(ops->process);
 	ret = ops->process(mod, sources, num_of_sources, sinks, num_of_sinks);
 
-	if (ret && ret != -ENOSPC && ret != -ENODATA) {
-		comp_err(dev, "module_process() error %d: for comp %d",
-			 ret, dev_comp_id(dev));
-		return ret;
+	if (ret) {
+		if (ret != -ENOSPC && ret != -ENODATA) {
+			comp_err(dev, "module_process() error %d: for comp %d",
+				 ret, dev_comp_id(dev));
+			return ret;
+		}
+
+		ret = 0;
```

Comment:
an alternative and much smaller change, achieving the same result - just `return 0` at line 303 / 307.

Reply:
yes, this would simplify. @softwarecki does this satisfy the requirement ?

Reply:
I agree. Even the first version of the code looked like this. Later I decided to do the same as in other places in the code. Let me change it.